src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java/JavaLanguageLexer.g4

```diff
@@ -112,16 +112,19 @@ tokens {
     PERMITS_CLAUSE, PATTERN_DEF, LITERAL_WHEN,
     RECORD_PATTERN_DEF, RECORD_PATTERN_COMPONENTS,
 
-    STRING_TEMPLATE_BEGIN, STRING_TEMPLATE_MID, STRING_TEMPLATE_END,
-    STRING_TEMPLATE_CONTENT, EMBEDDED_EXPRESSION_BEGIN, EMBEDDED_EXPRESSION,
-    EMBEDDED_EXPRESSION_END,
+    // We removed support for string template. String template tokens
+    // are no longer used, but we need to keep them to preserve token numbering,
+    // so that we are not breaking change with older versions.
+    // We used 'DEPRECATED' as prefix to indicate that they are not used.
+    DEPRECATED_STRING_TEMPLATE_BEGIN,DEPRECATED_STRING_TEMPLATE_MID, DEPRECATED_STRING_TEMPLATE_END,
```

Comment:
__Not__For_usage1__, __Not__For_usage2__
To completely forget why it was added.

I open for better names.

Reply:
good 
@nrmancuso what do you think

Reply:
We can do `NOT_FOR_USAGE_1` with comment that these are placeholders to maintain token ordering, we can skip mentioning string templates

Reply:
done